### PR TITLE
docs: 添加自适应布局demo 

### PR DESCRIPTION
**Pull request by stone-lyl**
### 👀 PR includes

✨ Feature

- [ ] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [ ] Solve the issue and close #0

🔧 Chore

- [ ] Test case
- [X] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| ❌      | ✅     |
因为demo 较多，都是通过 URL 方式引入的
image
因为自适应的使用问的也比较多，所以也一起写了
image

### 🔍 Self Check before Merge

- [ ] Add or update relevant Docs.
- [ ] Add or update relevant Demos.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: s2-site/examples/layout/adaptive/API.en.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+title: API
+order: 0
+---
+
+# Style
```
Comment (lijinke666) [inline s2-site/examples/layout/adaptive/API.en.md:6]: style ?

File: s2-site/examples/layout/adaptive/demo/meta.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  "title": {
+    "zh": "表格自适应",
+    "en": "Table Adaptive"
+  },
+  "demos": [
+    {
+      "filename": "window-adaptation.ts",
+      "title": {
+        "zh": "窗口自适应",
+        "en": "window adaptation"
```
Comment (lijinke666) [inline s2-site/examples/layout/adaptive/demo/meta.json:11]: 首字母大写?

File: s2-site/examples/layout/adaptive/demo/meta.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      },
+      "screenshot": "https://gw.alipayobjects.com/zos/antfincdn/yzBFZaMMLH/55a2d580-1121-468e-9d0b-25f3e9ae25df.png"
+    },
+    {
+      "filename": "container-adaptation.ts",
+      "title": {
+        "zh": "容器自适应",
+        "en": "container adaptation"
+      },
+      "screenshot": "https://gw.alipayobjects.com/zos/antfincdn/cEkDC%26g%24xj/df31b66d-4a76-4e69-be5a-f467af2d337d.png"
```
Comment (lijinke666) [inline s2-site/examples/layout/adaptive/demo/meta.json:21]: 截图换成 gif 吧, 直观点

File: s2-site/docs/manual/advanced/adaptive.zh.md [hunk 4]
```diff
@@ -202,3 +208,5 @@ const adaptive = {
   </div>
 </template>
 ```
+
+​📊 查看[Vue 组件自适应 demo](https://codesandbox.io/s/vue-adaptive-demo-4pptyy?file=/src/App.vue)
```
Comment (xingwanying) [inline s2-site/docs/manual/advanced/adaptive.zh.md:212]: /s2-site/docs/manual/basic/analysis/vue.zh.md

也加一下？

**Review by lijinke666 — APPROVED** [review]

**Pull request merged**